### DISCUSSION TITLE
How do I use this?

### DISCUSSION THREAD

**Discussion by FelineColt78602** [SteamRE/DepotDownloader#215] [Q&A]
I do not see the .bat file when I unzip this file, only a .sln file, am I missing something? I followed multiple videos on how to use this and none helped.

**Comment by FatalBulletHit** [comment]
## Requirements

### DepotDownloader
You need to download and extract the [latest release](https://github.com/SteamRE/DepotDownloader/releases/latest) of DepotDownloader.

### .NET 5.0
You need to have [.NET 5.0 Runtime](https://dotnet.microsoft.com/download/dotnet/5.0/runtime) installed.

### Steam account
You need a Steam account that owns the app (aka game) you want to download.

### App, depot and manifest ids
You need the app id and optionally a depot id and manifest id.
These can be easily found by searching for the app on [SteamDB](https://steamdb.info/).

- **App id**
The id of the app you want to download.
- **Depot id** [Optional]
The id of the depot you want to download. 
Depots are often things like expansions (aka DLCs), OS or language specific versions and so on.
- **Manifest id** [Optional]
The id of the manifest you want to download. 
This allows you to download older versions of an app's depot and requires a depot id.

## Making it do stuff
There are several ways you can use DepotDownloader, the easiest one is to open the shell of your choice (e.g. CMD, PowerShell or Bash) and navigate to where you extracted DepotDownloader to:
```
cd "C:\some\path\depotdownloader-2.4.1"
```
After that, start DepotDownloader with the parameters you desire.

I recommend you omit the password parameter as it complicates things when your password contains special characters.
DepotDownloader will prompt you to enter you password and potentially an authentication code on the first login if you have Steam Guard enabled.

**Examples**

Download Counter-Strike: Global Offensive (app id [730](https://steamdb.info/app/730/)):
```
.\DepotDownloader.exe -app 730 -username Gaben
```
Download the Arma 3 Creator DLC: Global Mobilization - Cold War Germany (depot id [1042220](https://steamdb.info/depot/1042220/)) of Arma 3 (app id [107410](https://steamdb.info/app/107410/)):
```
.\DepotDownloader.exe -app 107410 -depot 1042220 -username Miller
```
Download the German language pack (depot id [359555](https://steamdb.info/depot/359555/)) of Tom Clancy's Rainbow Six Siege (app id [359550](https://steamdb.info/app/359550/)) from patch 5.2.3 (manifest id [3439024125519410144](https://steamdb.info/depot/359555/history/?changeid=M:3439024125519410144)):
```
.\DepotDownloader.exe -app 359550 -depot 359555 -manifest 3439024125519410144 -username TomClancy
```

### Multiple app, depot and manifest ids
It can become quite tedious if you want to download multiple manifests, depots and/or apps, but you can automate this process, e.g. by using PowerShell. Why PowerShell? Because batch is bad and PowerShell comes pre-installed with Windows 10 and works on Linux, too.

1. Navigate to where you extracted DepotDownloader to.
2. Create a new file (Right click -> New -> Text Document) and rename it to `launchDepotDownloader.ps1` (make sure it has the `.ps1` extension)
3. Edit the file with the text editor of your choice or the ISE (Right click -> Edit) and paste the following:
```
# Username
$username = "username"

# Populate the $ids array, possible patterns: (appId), (appId, depotId) or (appId, depotId, manifestId)
# Example: Download app id 730, depot id 1042220 from app id 107410 and manifest 3439024125519410144 from depot id 359555 from app id 359550
# $ids = , (730), (107410, 1042220), (359550, 359555, 3439024125519410144)

$ids = , (730), (107410, 1042220), (359550, 359555, 3439024125519410144)



$ids | ForEach-Object {

    $appId, $depotId, $manifestId = $_

    $param = "-app $appId"

    if ($null -ne $depotId) {

        $param += " -depot $depotId"

        if ($null -ne $manifestId) {

            $param += " -manifest $manifestId"

        }
    }

    $param += " -username $username -remember-password"

    if ($null -ne $manifestId) {

        $param += " -dir .\depots\$depotId\$manifestId"

    }

    .\DepotDownloader.exe $param.Split(' ')

}
```
4. Adjust `$username` and the `$ids` array to your desire and save the file.
5. Right click the file > Run with PowerShell **\*** (potentially allow script execution)

**\*** If you don't have the option, you might have to install PowerShell from [here](https://github.com/PowerShell/PowerShell#get-powershell).
